librosa/util/utils.py

```diff
@@ -627,8 +632,26 @@ def normalize(S, norm=np.inf, axis=0):
     else:
         raise ParameterError('Unsupported norm: {}'.format(repr(norm)))
 
+    # get the vectors-ids smaller than the threshold
+    small_values_idx = (length < threshold)
+
     # Avoid div-by-zero
-    length[length < SMALL_FLOAT] = 1.0
+    length[small_values_idx] = 1.0
+
+    # init empty slice object
+    slices = [slice(None)] * S.ndim
+
+    # set slice for target-axis
+    target_axis_idx = int(np.logical_not(axis))
```

Comment:
> If we want to support n-dims, we need to introduce another parameter here to specify the axis where the we want to replace values.

I would expect the semantics to be that slices with low norm get set to the threshold on all axes other than `axis`.